### PR TITLE
Elaborate on definition for Conformance tests

### PR DESCRIPTION
**Pull request by xpivarc**
### What this PR does
In order to promote usage of Conformance tests but also the extension of it, we define some goals and requirements.

Fixes #

Links to places where the discussion took place: 

### Release note

```release-note
NONE
```

### CODE REVIEW

File: docs/conformance.md [hunk 1]
```diff
@@ -1,14 +1,44 @@
...
+
+The tests are always part of Kubevirt e2e tests that are usually approved by individual Sigs. For the Conformance tests we require higher requirments than than 
+regular tests, most significant requirments are:
+- Test is not Disruptive
+- Test is not flaky and is reliable per data available from CI team
+- For more concreate details see below
+
+
+## Conformance tests requirments:
+- The feature being tested is GA (no feature gated)
```
Comment (dankenigsberg) [inline docs/conformance.md:20]:
```suggestion
- The feature being tested is GA (no feature gate)
```

File: docs/conformance.md [hunk 2]
```diff
@@ -1,14 +1,44 @@
...
+- The feature is not deprecated
+- The feature is turned on by default (no feature toogle)
```
Comment (fossedihelm) [inline docs/conformance.md:22]: An exception to this could be upgrade test. WDYT?

File: docs/conformance.md [hunk 3]
```diff
@@ -1,14 +1,44 @@
 # Conformance testing
 
+The Conformance tests are subset of our e2e test suite that defines set of features
+defining Kubevirt core functionality. The goal of the suite is to make sure that selected features work across different Vendors, CNIs, CSIs, CRIs implementations.
+
+## How are Conformance tests selected
+
+By default tests with a `[Conformance]` tag (and from release 0.59 also decorator) will be executed and tests with a
```
Comment (fossedihelm) [inline docs/conformance.md:8]: How about adding the label decorator?

File: docs/conformance.md [hunk 4]
```diff
@@ -1,14 +1,44 @@
...
+`[Disruptive]` tag will be skipped. These tests will not destroy or modify
+anything outside of the explicitly created test namespaces by the binary.
+
+The tests are always part of Kubevirt e2e tests that are usually approved by individual Sigs. For the Conformance tests we require higher requirments than than
```
Comment (dhiller) [inline docs/conformance.md:12]:
```suggestion
The tests are always part of Kubevirt e2e tests that are usually approved by individual Sigs. For the Conformance tests we have higher requirements than the 
```

File: docs/conformance.md [hunk 5]
```diff
@@ -1,14 +1,44 @@
...
+- The feature is not deprecated
+- The feature is turned on by default (no feature toogle)
+- The test uses only GAed Kubernetes APIs/features
+- The test uses only APIs available to user
+- The test doesn't use direct access to virt-launcher/libvirt
+- It doesn't require any special hardware (such as gpu/pci devices,...) being available on the cluster other than the default required device nodes (kvm, virtio-net)
+- It doesn't use `Skip` or any other helper that uses it
+- It does not require access to public network, all images or resources should be able to be pre-pulled
+- It does not rely on environment of the executor
+- It has history of stability and is not flaky
```
Comment (dhiller) [inline docs/conformance.md:30]: nit

```suggestion
- It has history of stability and is not flaky, also it doesn't have the QUARANTINE label.
```

File: docs/conformance.md (hunk 3 above)
Comment (xpivarc) [reply docs/conformance.md:8]: What do you mean by adding?

File: docs/conformance.md [hunk 6]
```diff
@@ -1,14 +1,44 @@
 # Conformance testing
 
+The Conformance tests are subset of our e2e test suite that defines set of features
+defining Kubevirt core functionality. The goal of the suite is to make sure that selected features work across different Vendors, CNIs, CSIs, CRIs implementations.
+
+## How are Conformance tests selected
+
+By default tests with a `[Conformance]` tag (and from release 0.59 also decorator) will be executed and tests with a
+`[Disruptive]` tag will be skipped. These tests will not destroy or modify
```
[the remark this answers is not in the document]
Comment (xpivarc) [reply docs/conformance.md:9]: I would keep this as documentation only for now

File: docs/conformance.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (xpivarc) [reply docs/conformance.md:12]: Sounds good

File: docs/conformance.md (hunk 3 above)
Comment (fossedihelm) [reply docs/conformance.md:8]: I mean, here we are only mentioning the `[Conformance]` tag.I think we want to switch using the Conformance label decorator.

File: docs/conformance.md (hunk 2 above)
Comment (xpivarc) [reply docs/conformance.md:21]: I am not sure upgrade tests are eligible as they do modify the installation. I believe that has to be leave for vendors, users to tests for themself...

File: docs/conformance.md (hunk 3 above)
Comment (xpivarc) [reply docs/conformance.md:8]: Right, as we don't have the label yet I would leave this out and fill in later.

**Comment by xpivarc** [comment]
@dankenigsberg @fossedihelm @dhiller PTAL

File: docs/conformance.md (hunk 3 above)
Comment (dhiller) [reply docs/conformance.md:8]: Part of another PR: https://github.com/kubevirt/kubevirt/pull/12682/files#diff-2be6ff8bee7a7b41764cd824e3e7afbc1ae3facd2893bdeb60f7ac978303a387R8

**Comment by fossedihelm** [comment]
/lgtm
Thanks

File: docs/conformance.md [hunk 7]
```diff
@@ -1,14 +1,43 @@
...
+anything outside of the explicitly created test namespaces by the binary.
+
+The tests are always part of Kubevirt e2e tests that are usually approved by individual [Sigs](https://github.com/kubevirt/community/blob/main/sig-list.md#special-interest-groups). For the Conformance tests we have higher requirments than the regular tests, most significant requirments are:
```
Comment (jean-edouard) [inline docs/conformance.md:12]:
```suggestion
The tests are always part of Kubevirt e2e tests that are usually approved by individual [Sigs](https://github.com/kubevirt/community/blob/main/sig-list.md#special-interest-groups). For the Conformance tests we have higher requirements than the regular tests, most significant requirements are:
```

File: docs/conformance.md [hunk 8]
```diff
@@ -1,14 +1,43 @@
...
+- Test is not Disruptive
+- Test is not flaky and is reliable per data available from CI team
+- For more concreate details see below
```
Comment (jean-edouard) [inline docs/conformance.md:15]:
```suggestion
- For more concrete details see below
```

File: docs/conformance.md [hunk 9]
```diff
@@ -1,14 +1,43 @@
...
+
+
+## Conformance tests requirments:
```
Comment (jean-edouard) [inline docs/conformance.md:18]:
```suggestion
## Conformance tests requirements:
```

**Review by jean-edouard — COMMENTED** [review]
/approve
/hold
Thank you for this, just found a couple typos, feel free to unhold once they're addressed!

File: docs/conformance.md [hunk 10]
```diff
@@ -1,14 +1,43 @@
 # Conformance testing
 
+The Conformance tests are subset of our e2e test suite that defines set of features
+defining Kubevirt core functionality. The goal of the suite is to make sure that selected features work across different Vendors, CNIs, CSIs, CRIs implementations.
```
Comment (jean-edouard) [inline docs/conformance.md:4]:
```suggestion
The Conformance tests are a subset of our e2e test suite that cover a set of features
defining Kubevirt core functionalities. The goal of the suite is to make sure that selected features work across different Vendors, CNIs, CSIs, CRIs implementations.
```

File: docs/conformance.md [hunk 11]
```diff
@@ -1,14 +1,43 @@
...
+- Test is not flaky and is reliable per data available from CI team
+- For more concreate details see below
+
+
+## Conformance tests requirments:
+- The feature being tested is GA (no feature gate)
+- The feature is not deprecated
+- The feature is turned on by default (no feature toogle)
+- The test uses only GAed Kubernetes APIs/features
+- The test uses only APIs available to user
```
Comment (lyarwood) [inline docs/conformance.md:23]: Should the eventual goal here to also only use GA'd KubeVirt features without FGs? I appreciate that isn't possible right now so the current wording is fine but would be nice to document something here.

Comment (xpivarc) [reply docs/conformance.md:23]: See first point "- The feature being tested is GA (no feature gate)"

**Comment by xpivarc** [comment]
@jean-edouard PTAL

File: docs/conformance.md (hunk 11 above)
Comment (lyarwood) [reply docs/conformance.md:23]: ACK apologies missed that.

Comment (xpivarc) [reply docs/conformance.md:23]: Thank you for the review. I can add the reasoning if some points need to.

**Comment by jean-edouard** [comment]
/unhold
Note: the hold was just to give you time to fix the typos I found. In such trivial cases, I think it's fine for the author to unhold without waiting for the approver.

**Comment by xpivarc** [comment]
> /unhold Note: the hold was just to give you time to fix the typos I found. In such trivial cases, I think it's fine for the author to unhold without waiting for the approver.

I still like to leave option for the author of the hold to check everything is as asked and nothing new pop out

File: docs/conformance.md (hunk 2 above)
Comment (fossedihelm) [reply docs/conformance.md:21]:
```suggestion
- The feature is turned on by default (no feature toggle)
```

**Comment by fossedihelm** [comment]
@lyarwood Do you want to have another look?
It looks good from my point of view, but want to be sure you are ok before it gets merged

**Comment by lyarwood** [comment]
Makes sense to me , thanks!

/lgtm

File: docs/conformance.md [hunk 12]
```diff
@@ -1,14 +1,43 @@
...
+- It has history of stability and is not flaky, and consequently doesn't have the QUARANTINE label.
+- It doesn't rely on specific implementation of CRI, CSI, CNI
+
+Exceptions are allowed but must be explicit and reasoned.
+
+## Delivery
+
 KubeVirt releases a `kubevirt/conformance:<release>` image and
 `conformance.yaml` manifest starting from `v0.33.0`. It can be executed as a
 [Sonobuoy](https://sonobuoy.io/) plugin to verify if the underlying cluster
```
Comment (brianmcarey) [inline docs/conformance.md:38]: Just a question outside the scope of this PR. 

Do we still want to use sonobuoy for executing these conformance tests? I think we see some limitations there like we are currently not running the tests in parallel. 

Thanks for this @xpivarc  - it should be a great help.

Comment (xpivarc) [reply docs/conformance.md:38]: I would say we should follow Kubernetes here to give the same experience.

**Pull request merged**